front/app/containers/Admin/formBuilder/components/FormBuilderSettings/index.tsx

```diff
@@ -130,6 +131,7 @@ const FormBuilderSettings = ({ field, onDelete, onClose }: Props) => {
           iconColor={colors.error}
           onClick={() => onDelete(field.index)}
           minWidth="160px"
+          data-cy="e2e-delete-field"
         >
```

Comment:
🤔 I think the opposite is true. `data-cy` is safer because it is isolated from many changes and as such when the tests fail, there is a high likelihood that something doesn't work. The challenge with contains is that it is tied into the copy. The test will fail when the copy changes. Tracking a change in copy and failing tests is not ideal. More information here https://docs.cypress.io/guides/references/best-practices

<img width="797" alt="image" src="https://user-images.githubusercontent.com/12659000/195127267-dd1813c4-3321-4453-a298-7a924d9d36b7.png">

Reply:
Interesting - just reading a bit more. I was under the impression that we didn't want to use them, but it seems like it's best practice to? Should I also be using these over cy.contains() selectors then?

![image](https://user-images.githubusercontent.com/33987955/195127651-7b3cf2af-f82e-4328-b182-45e7b2b7c719.png)

Reply:
Yes, it is something I was hoping to gradually introduce but I'm glad you've seen it

Reply:
Sounds good :) this makes sense to me. I'll update my other active PR tomorrow to use less brittle selectors.